tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15371,16 +15372,11 @@ void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr, bool isA
 
 // if this is the Entry FD, then try adding the target profile
 // shader attribute to the FD and carry on with validation
-void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD, bool &isActiveEntry) {
+void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
+                                       bool &isActiveEntry) {
   isActiveEntry = false;
```

Comment:
We need a comment to explain this variable, I think.  For example:
> When isActiveEntry is true and this function is an entry point, this entry point is used in compilation.  This is an important distinction when diagnosing certain types of errors based on the compilation parameters.  For example, if isActiveEntry is false, diagnostics dependent on the shader model should not be performed.  That way we won't raise an error about a feature used by the inactive entry that's not available in the current shader model.  Since that entry point is not used, it may still be valid in another compilation where a different shader model is specified.